File: JS/TeoremaDaDivisaoEuclidiana.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

let line = gets().split(" ");
let ar = getQuoAndRest( line[0], line[1] );

console.log( `${ar[0]} ${ar[1]}` );

/**
 * Get the quocient and rest of the division.
 */
function getQuoAndRest(a = 0, b = 0) {
    let q = parseInt( a / b );
    let r = a - (q * b);

    if ( r < 0 ) {
        r += Math.abs(b);

        q = (a-r) / b;
    }
    
    return [q, r];
}
module.exports = {getQuoAndRest}

File: GO/TeoremaDaDivisaoEuclidiana.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "math"

/**
 * Get the quocient and rest of the division.
 */
func getQuoAndRest(a int, b int) []int {
	q := a / b
	r := a % b

	if r < 0 {
		r += int(math.Abs(float64(b)))

		q = (a - r) / b
	}

	return []int{q, r}
}
